### PR TITLE
Close gnucash with mouse click

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
An sporadic race condition happened when the loop iterated faster than the display updates.

**gnucash** is a GUI test, mouse clicks should be ok.

- Related ticket: https://progress.opensuse.org/issues/40319
- **Please merge needles first**: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/439

## Verification run:

|                      | gnome             | kde                  | xfce                 | lxde                  |
| ---------------- | ------------------ | ------------------- | ------------------ | ------------------ |
| Tumbleweed | [tw-gnome](http://slindomansilla-vm.qa.suse.de/tests/94) | [10x tw-kde](http://slindomansilla-vm.qa.suse.de/tests/overview?build=poo_40319-gnucash) (4 of 10 failing on **ensure_installed**. See [PR#5830](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/5830)) | [tw-xfce](http://slindomansilla-vm.qa.suse.de/tests/108) (Fails on **ensure_installed**, see [PR#5830](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/5830)) | [tw-lxde](http://slindomansilla-vm.qa.suse.de/tests/111) |
| Leap 15.1     | [leap15.1-gnome](http://slindomansilla-vm.qa.suse.de/tests/96) | [leap15.1-kde](http://slindomansilla-vm.qa.suse.de/tests/97) | [leap15.1-xfce](http://slindomansilla-vm.qa.suse.de/tests/115) | N/A |
| Leap 15        | [leap15.0-gnome](http://slindomansilla-vm.qa.suse.de/tests/117) | [leap15.0-kde](http://slindomansilla-vm.qa.suse.de/tests/122) | [leap15.0-xfce](http://slindomansilla-vm.qa.suse.de/tests/133) | N/A |
| Leap 42.3     | (coming soon) | (coming soon) | (coming soon) | N/A |

### CODE REVIEW

**Comment by SergioAtSUSE** [comment]
Related to: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/5745

File: tests/x11/gnucash.pm [hunk 1]
```diff
@@ -32,16 +33,9 @@ sub run {
...
-    for (1 .. 7) {
-        assert_screen [qw(test-gnucash-1 gnucash gnucash-save-changes generic-desktop)];
-        last              if match_has_tag 'generic-desktop';
-        send_key 'alt-c'  if match_has_tag 'test-gnucash-1';
-        send_key 'alt-w'  if match_has_tag 'gnucash-save-changes';
-        send_key 'ctrl-q' if match_has_tag 'gnucash';
-    }
+    assert_and_click('gnucash-close-window');
+    assert_and_click('gnucash-close-without-saving-changes') if (is_leap('<=15.1'));
+    assert_screen('generic-desktop');
```
Comment (okurz) [inline tests/x11/gnucash.pm]: This is already ensured by the post_run_hook, see the double check for generic-desktop http://10.160.66.74/tests/913#step/gnucash/22

Comment (SergioAtSUSE) [reply tests/x11/gnucash.pm]: If this module is not able to close the window, I want it to fail, not making the next module fail.

Comment (okurz) [reply tests/x11/gnucash.pm]: Yeah I think the post_run_hook will do exactly that but you can try to prove me wrong :)

**Comment by SergioAtSUSE** [comment]
@okurz, sorry, mind game, I misread as post_fail_hook. Done.

**Comment by SergioAtSUSE** [comment]
@rwx788,
You mean the retries on my verification runs?
That is because of the problems of my webui, nothing to do with the code changes.
And this is also the reason why I could not put a 10 rounds of runs for each case (as I wish)

**Comment by rwx788** [comment]
we don't need 10 runs for each, 10 runs for one is more than enough. I've just expressed my concern. And you already did 7 runs, which is a good sign. If you also have same concern, let's trigger more runs, if not, let's merge and see what will happen =)

**Comment by SergioAtSUSE** [comment]
Well right, at least 10 run for one and 1 run for each of the other cases should be enough.

File: lib/susedistribution.pm [hunk 2]
```diff
@@ -276,6 +276,7 @@ sub ensure_installed {
         die "pkcon install did not succeed, return code: $ret";
     }
     send_key("alt-f4");    # close xterm
+    assert_screen('generic-desktop');
```
Comment (okurz) [inline lib/susedistribution.pm]: if you want to do that please do it as a separate PR to make that change more obvious as it will have a big impact

File: tests/x11/gnucash.pm [hunk 3]
```diff
@@ -32,16 +33,8 @@ sub run {
...
-    # arbitrary limit
-    for (1 .. 7) {
-        assert_screen [qw(test-gnucash-1 gnucash gnucash-save-changes generic-desktop)];
-        last              if match_has_tag 'generic-desktop';
-        send_key 'alt-c'  if match_has_tag 'test-gnucash-1';
-        send_key 'alt-w'  if match_has_tag 'gnucash-save-changes';
-        send_key 'ctrl-q' if match_has_tag 'gnucash';
-    }
+    assert_and_click('gnucash-close-window');
+    assert_and_click('gnucash-close-without-saving-changes') if (is_leap('<=15.1'));
```
Comment (okurz) [inline tests/x11/gnucash.pm]: leap 15.1 really is different to TW?

File: lib/susedistribution.pm (hunk 2 above)
Comment (SergioAtSUSE) [reply lib/susedistribution.pm]: That's funny, I thought you would complain that I didn't wanted to do it on another PR blocking my current PR and ticket, and then I decided to include it on this PR.

But I have put it on a different commit so it can be reverted. Do you really just want another PR or another ticket?

File: tests/x11/gnucash.pm (hunk 3 above)
Comment (SergioAtSUSE) [reply tests/x11/gnucash.pm]: Yes :) http://slindomansilla-vm.qa.suse.de/tests/93#step/gnucash/22
